src/main/java/converters/IntConverter.java

```diff
@@ -0,0 +1,178 @@
+	 *                    if none is provided, {@link IntOption#BIG} is used)
+	 * @return the unsigned integer value string
+	 * @throws ConverterException if the input binary string is invalid
+	 */
+	@Override
+	public String convertTo(String bitStr, IOption[] options) throws ConverterException {
+		validateInput(bitStr, "^[01]+$");
+		IntOption endian = getEndianFromOptions(options);
+
+		String updatedBitStr = processBitString(bitStr, endian);
+
+		String decimalNumber = STRING_WITH_ZERO;
+		String powerOfTwo = "1";
+		for (int i = updatedBitStr.length() - 1; i >= 0; i--) {
+			int digit = updatedBitStr.charAt(i) - '0';
+			if (digit == 0) {
+				powerOfTwo = multiplyStrings(powerOfTwo, "2");
+				continue;
+			}
+
+			String product = multiplyStrings(String.valueOf(digit), powerOfTwo);
+			decimalNumber = addStrings(decimalNumber, product);
+			powerOfTwo = multiplyStrings(powerOfTwo, "2");
+		}
+		return decimalNumber;
+	}
+
+	/**
+	 * Converts an unsigned integer value string to a binary string.
+	 *
+	 * @param input    the unsigned integer value string to convert
+	 * @param options  an array of {@link IntOption} options (first is taken,
+	 *                    if none is provided, {@link IntOption#BIG} is used)
+	 * @return the binary string
+	 * @throws ConverterException if the input unsigned integer value string is invalid
+	 */
+	@Override
+	public String convertFrom(String input, IOption[] options) throws ConverterException {
+		validateInput(input, "^\\d+$");
+		IntOption endian = getEndianFromOptions(options);
+
+		StringBuilder binary = new StringBuilder();
+		StringBuilder updatedInput = new StringBuilder(input);
+
+		while (updatedInput.length() > 0) {
+			StringBuilder semiResult = new StringBuilder();
+			boolean foundNoneZero = false;
+			int remainder = 0;
+
+			for (int i = 0; i < updatedInput.length(); i++) {
+				int digit = Character.getNumericValue(updatedInput.charAt(i));
+				int dividend = remainder * TEN + digit;
+				char charDigit = (char) ('0' + (dividend / 2));
+				remainder = dividend % 2;
+
+				if (!foundNoneZero && charDigit == '0'){
+					continue;
+				}
+				foundNoneZero = true;
+				semiResult.append(charDigit);
+			}
+			binary.append(remainder);
+			updatedInput = semiResult;
+		}
+
+		String bitStr = binary.reverse().toString();
+		bitStr = processBitString(bitStr, endian);
+
+		return bitStr;
+	}
+
+	private static String addStrings(String firstStrValue, String secondStrValue) {
+		StringBuilder sb = new StringBuilder();
+		int carry = 0;
+		int i = firstStrValue.length() - 1;
+		int j = secondStrValue.length() - 1;
+		while (i >= 0 || j >= 0 || carry > 0) {
+			int sum = carry;
+			if (i >= 0) {
+				sum += firstStrValue.charAt(i--) - '0';
+			}
+			if (j >= 0) {
+				sum += secondStrValue.charAt(j--) - '0';
+			}
+			carry = sum / TEN;
+			sb.append(sum % TEN);
+		}
+		return sb.reverse().toString();
+	}
+
+	private static String multiplyStrings(String firstStrValue, String secondStrValue) {
```

Comment:
**suggestion:** I am missing some context why this function exists and needs to be used, a comment would help